diff --git a/compareFirmware/firmware_example/Makefile b/compareFirmware/firmware_example/Makefile
index b640d68f7e..a93e59197a 100644
--- a/compareFirmware/firmware_example/Makefile
+++ b/compareFirmware/firmware_example/Makefile
@@ -15,12 +15,12 @@ FEATURES_REQUIRED += periph_gpio_irq
 # USB Ethernet and gnrc_networking
 GNRC_NETIF_NUMOF := 1
 
-USEMODULE += gnrc_netdev_default
-USEMODULE += auto_init_gnrc_netif
-USEMODULE += gnrc_ipv6
-USEMODULE += gnrc_icmpv6_error
-USEMODULE += gnrc_udp
-USEMODULE += gnrc_icmpv6_echo
+#USEMODULE += gnrc_netdev_default
+#USEMODULE += auto_init_gnrc_netif
+#USEMODULE += gnrc_ipv6
+#USEMODULE += gnrc_icmpv6_error
+#USEMODULE += gnrc_udp
+#USEMODULE += gnrc_icmpv6_echo
 
 USEMODULE += ps
 USEMODULE += netstats_l2
