diff --git a/compareFirmware/firmware_example/Makefile b/compareFirmware/firmware_example/Makefile
index ef471cdfd6..82db002d82 100644
--- a/compareFirmware/firmware_example/Makefile
+++ b/compareFirmware/firmware_example/Makefile
@@ -12,6 +12,10 @@ USEMODULE += shell_commands
 FEATURES_REQUIRED += periph_adc
 FEATURES_REQUIRED += periph_gpio_irq
 
+# acceleration sensor
+#USEMODULE += lis2dh12_spi
+USEMODULE += lis3dh
+
 # gnrc_networking
 GNRC_NETIF_NUMOF := 1
 
diff --git a/compareFirmware/firmware_example/main.c b/compareFirmware/firmware_example/main.c
index 8389ec7cb2..6e6e59b79c 100644
--- a/compareFirmware/firmware_example/main.c
+++ b/compareFirmware/firmware_example/main.c
@@ -26,6 +26,14 @@
 #include "periph/adc.h"
 #include "board.h"
 
+#include "lis3dh.h"
+#include "lis3dh_params.h"
+/* allocate device descriptor */
+static lis3dh_t dev;
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
 #define ADC_SAMPLE_LINE 0
 
 #define MAIN_QUEUE_SIZE (8)
@@ -54,10 +62,43 @@ void button_int_cb(void* arg){
     puts("interrupt received.");
 }
 
+/* LIS3DH init function */
+void lis3dh_func_init(void) {
+
+    /* init lis */
+    if (lis3dh_init(&dev, &lis3dh_params[0]) == 0) {
+        puts("lis3dh [Initialized]");
+    }
+    else {
+        puts("lis3dh [Failed]");
+    }
+
+    lis3dh_set_odr(&dev, lis3dh_params[0].odr);
+    lis3dh_set_scale(&dev, lis3dh_params[0].scale);
+    lis3dh_set_axes(&dev, LIS3DH_AXES_XYZ);
+
+}
+
+/* read LIS values */
+static int lis3dh_read(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    lis3dh_data_t data;
+
+    lis3dh_read_xyz(&dev, &data);
+
+    printf("X: %d  Y: %d  Z: %d\n", data.acc_x, data.acc_y, data.acc_z);
+
+    return 0;
+}
+
 /* shell commands */
 static const shell_command_t shell_commands[] = {
-    { "toggle_LED", "toggles the LED status of LED0", toggle_LED},
-    { "adc", "reads the ADC0", read_ADC},
+    { "toggle_LED", "toggles the LED status of LED0", toggle_LED },
+    { "adc", "reads the ADC0", read_ADC },
+    { "lis_read", "Read acceleration data", lis3dh_read },
     { NULL, NULL, NULL }
 };
 
@@ -81,6 +122,9 @@ int main(void)
     /* initializes the message queue */
     msg_init_queue(_main_msg_queue, MAIN_QUEUE_SIZE);
 
+    /* init acceleration sensor */
+    lis3dh_func_init();
+
     /* start shell */
     puts("All up, running the shell now");
     char line_buf[SHELL_DEFAULT_BUFSIZE];
